doc/stdenv/platform-notes.chapter.md

```diff
@@ -46,22 +46,195 @@ Some common issues when packaging software for Darwin:
   }
   ```
 
-- Some packages assume xcode is available and use `xcrun` to resolve build tools like `clang`, etc. This causes errors like `xcode-select: error: no developer tools were found at '/Applications/Xcode.app'` while the build doesn’t actually depend on xcode.
+- Some packages assume Xcode is available and use `xcrun` to resolve build tools like `clang`, etc. The Darwin stdenv includes `xcrun`, and it will return the path to any binary available in a build.
 
   ```nix
   stdenv.mkDerivation {
     name = "libfoo-1.2.3";
     # ...
+    nativeBuildInputs = [ bison ];
+    buildCommand = ''
+      xcrun bison foo.y # produces foo.tab.c
+      # ...
+    '';
+  }
+  ```
+  The package `xcbuild` can be used to build projects that really depend on Xcode. However, this replacement is not 100% compatible with Xcode and can occasionally cause issues.
+
+  Note: Some packages may hardcode an absolute path to `xcrun`, `xcodebuild`, or `xcode-select`. Those paths should be removed or replaced.
+
+  ```nix
+  stdenv.mkDerivation {
+    name = "libfoo-1.2.3";
     prePatch = ''
       substituteInPlace Makefile \
-          --replace-fail '/usr/bin/xcrun clang' clang
+        --replace-fail /usr/bin/xcrun xcrun
+        # or: --replace-fail /usr/bin/xcrun '${lib.getExe' buildPackages.xcbuild "xcrun"}'
     '';
   }
   ```
 
-  The package `xcbuild` can be used to build projects that really depend on Xcode. However, this replacement is not 100% compatible with Xcode and can occasionally cause issues.
+- When a package indicates a minimum supported version, also called the deployment target, you can set it in your derivation using `darwinMinVersionHook`.
+
+  The deployment target controls how Darwin handles availability and access to some APIs. In most cases, if a deployment target is newer than the first availability of an API, that API will be linked directly. Otherwise, the API will be weakly linked and checked at runtime. On aarch64-darwin, the minimum version cannot be lower than 11.0. If a lower version is set, 11.0 will be used instead.
```

Comment:
Probably instead of mentioning `aarch64-darwin` here we should just say that the highest deployment target wins and that the default is the Nixpkgs baseline for the platform. Putting SDK documentation first means that we could just say it works like SDKs here.